RDKB-59244: WPS Enhancement task

SanjayVenkatesan190 opened the pull request:
Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure: 
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by:sanjayvenkatesan1902@gmail.com

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Fixed merge conflict.

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjayvenkatesan1902@gmail.com

Commit:
RDKB-36610: WPS Enhancement task

    Impacted Platforms: All RDKB platforms.

    Reason for change: Fixed merge conflict.

    Test Procedure:

    Risks: Medium

    Priority: P1

    Signed-off-by:sanjayvenkatesan1902@gmail.com

Commit:
RDKB-36610: WPS Enhancement task

        Impacted Platforms: All RDKB platforms.

        Reason for change: Fixed merge conflict.

        Test Procedure:

        Risks: Medium

        Priority: P1

        Signed-off-by:sanjayvenkatesan1902@gmail.com

Commit:
RDKB-36610: WPS Enhancement task

        Impacted Platforms: All RDKB platforms.

        Reason for change:

        Test Procedure:

        Risks: Medium

        Priority: P1

        Signed-off-by:sanjayvenkatesan1902@gmail.com

Commit:
RDKB-36610: WPS Enhancement task

        Impacted Platforms: All RDKB platforms.

        Reason for change:

        Test Procedure:

        Risks: Medium

        Priority: P1

        Signed-off-by:sanjayvenkatesan1902@gmail.com

Commit:
RDKB-36610: WPS Enhancement task(CONF)

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

Commit:
RDKB-36610: WPS Enhancement task(CONF)

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

SanjayVenkatesan190 committed:
Merge branch 'rdkcentral:develop' into WPS_DISTRO_FLAG

Commit:
RDKB-36610: WPS Enhancement task(CONF)

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com# Please enter a commit message to explain why this merge is necessary,

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

SanjayVenkatesan190 committed:
Merge branch 'rdkcentral:develop' into WPS_DISTRO_FLAG

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

SanjayVenkatesan190 committed:
Merge branch 'rdkcentral:develop' into WPS_DISTRO_FLAG

SanjayVenkatesan190 committed:
Merge branch 'rdkcentral:develop' into WPS_DISTRO_FLAG

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjay_venkatesan@comcast.com

Commit:
RDKB-36610: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change:

Test Procedure:

Risks: Medium

Priority: P1

Signed-off-by:sanjayvenkatesan1902@gmail.com

SanjayVenkatesan190 committed:
Merge branch 'rdkcentral:develop' into WPS_DISTRO_FLAG

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/OneWifi?pullRequest=241) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/OneWifi?pullRequest=241) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: SanjayVenkatesan190<br/>:x: Sanjay Venkatesan<br/><hr/>**Sanjay Venkatesan** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/OneWifi?pullRequest=241) it.</sub>

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -806,6 +810,7 @@ void callback_Wifi_VAP_Config(ovsdb_update_monitor_t *mon,
         struct schema_Wifi_VAP_Config *old_rec,
         struct schema_Wifi_VAP_Config *new_rec)
 {
+    wifi_util_info_print(WIFI_DB,"SJY %s:%d\n", __func__, __LINE__);
```
remove debug log

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -993,9 +998,14 @@ void callback_Wifi_VAP_Config(ovsdb_update_monitor_t *mon,
             l_bss_param_cfg->wepKeyLength = new_rec->wep_key_length;
             l_bss_param_cfg->bssHotspot = new_rec->bss_hotspot;
+#if defined(FEATURE_SUPPORT_WPS)
+            wifi_util_info_print(WIFI_DB,"SJY %s:%d: WPS support enabled\n", __func__, __LINE__);
```
remove debug log

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -993,9 +998,14 @@ void callback_Wifi_VAP_Config(ovsdb_update_monitor_t *mon,
             l_bss_param_cfg->wpsPushButton = new_rec->wps_push_button;
             l_bss_param_cfg->wps.methods = new_rec->wps_config_methods;
             l_bss_param_cfg->wps.enable = new_rec->wps_enabled;
+            wifi_util_info_print(WIFI_DB,"SJY %s:%d: WPS push button %d config methods %d wps enabled %d\n",
```
remove debug log

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -2551,15 +2563,17 @@ int wifidb_update_wifi_vap_info(char *vap_name, wifi_vap_info_t *config,
         cfg.wep_key_length = config->u.bss_info.wepKeyLength;
         cfg.bss_hotspot = config->u.bss_info.bssHotspot;
+#ifdef FEATURE_SUPPORT_WPS
+        wifi_util_info_print(WIFI_DB, "SJY %s:%d: WPS is supported\n", __func__, __LINE__);
```
remove debug logs

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -2551,15 +2563,17 @@ int wifidb_update_wifi_vap_info(char *vap_name, wifi_vap_info_t *config,
-#else
         cfg.wps_enabled = config->u.bss_info.wps.enable;
-#endif /* _CBR2_PRODUCT_REQ_ */
+        wifi_util_info_print(WIFI_DB,
```
remove debug log

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -4482,6 +4500,7 @@ static void wifidb_vap_config_ext(wifi_vap_info_map_t *config, rdk_wifi_vap_info
 ********************************************** ****************************************/
 void wifidb_vap_config_correction(wifi_vap_info_map_t *l_vap_map_param)
 {
+    wifi_util_info_print(WIFI_DB, "SJY Entered %s:%d:\n", __func__, __LINE__);
```
remove debug log

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -5727,9 +5751,15 @@ int wifidb_get_wifi_vap_info(char *vap_name, wifi_vap_info_t *config,
             config->u.bss_info.wpsPushButton = pcfg->wps_push_button;
             config->u.bss_info.wps.methods = pcfg->wps_config_methods;
             config->u.bss_info.wps.enable = pcfg->wps_enabled;
+            wifi_util_info_print(WIFI_DB,
```
remove debug log

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -6594,10 +6626,12 @@ int wifidb_init_vap_config_default(int vap_index, wifi_vap_info_t *config,
-            cfg.u.bss_info.wps.enable = false;
-#else
+#ifdef FEATURE_SUPPORT_WPS
+ wifi_util_info_print(WIFI_DB, "SJY %s:%d: Setting WPS enable as true as its supported\n", __func__, __LINE__);
```
remove debug logs

bogdanbogush2 commented on source/db/wifi_db_apis.c:
```diff
@@ -7574,6 +7610,7 @@ int wifi_db_update_global_config(wifi_global_param_t *global_cfg)
 
 #ifndef NEWPLATFORM_PORT
     memset(strValue, 0, sizeof(strValue));
+    #ifdef FEATURE_SUPPORT_WPS
```
remove indentation

bogdanbogush2 commented on source/dml/tr_181/ml/cosa_wifi_dml.c:
```diff
@@ -11764,34 +11765,26 @@ WPS_SetParamBoolValue
-#else
         if (vapInfo->u.bss_info.wps.enable != bValue) {
-            wifi_util_dbg_print(WIFI_DMCLI, "%s:%d:key=%d bValue=%d  \n", __func__, __LINE__,
+            wifi_util_info_print(WIFI_DMCLI, "SJY %s:%d:key=%d bValue=%d\n", __func__, __LINE__,
```
remove debug log

bogdanbogush2 commented on source/dml/tr_181/ml/cosa_wifi_dml.c:
```diff
@@ -11764,34 +11765,26 @@ WPS_SetParamBoolValue
+            wifi_util_info_print(WIFI_DMCLI, "SJY %s:%d:key=%d bValue=%d\n", __func__, __LINE__,
                 vapInfo->u.bss_info.wps.enable, bValue);
             vapInfo->u.bss_info.wps.enable = bValue;
+            wifi_util_info_print(WIFI_DMCLI, "SJY %s:%d:Value of wps enable is %d\n", __func__, __LINE__,
```
remove debug log

bogdanbogush2 commented on source/webconfig/wifi_ovsdb_translator.c:
```diff
@@ -3872,8 +3883,11 @@ webconfig_error_t translate_ovsdb_to_vap_info_common(const struct schema_Wifi_VI
     vap->u.bss_info.bssTransitionActivated = vap_row->btm;
     vap->u.bss_info.nbrReportActivated = vap_row->rrm;
+#ifdef FEATURE_SUPPORT_WPS
+    wifi_util_error_print(WIFI_WEBCONFIG,"SJY %s:%d: The value of wps is %d\n", __func__, __LINE__, vap_row->wps);
```
remove debug log

bogdanbogush2 commented on source/webconfig/wifi_ovsdb_translator.c:
```diff
@@ -2699,11 +2707,13 @@ webconfig_error_t translate_vap_info_to_vif_state_common(const wifi_vap_info_t *
-
+#ifdef FEATURE_SUPPORT_WPS
     if (vap->u.bss_info.wps.enable) {
+        wifi_util_error_print(WIFI_WEBCONFIG,"%s:%d: WPS is enabled for vap\n", __func__, __LINE__);
```
remove debug log

SanjayVenkatesan190 committed:
Merge branch 'rdkcentral:develop' into WPS_DISTRO_FLAG

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

Commit:
Merge branch 'develop' into WPS_DISTRO_FLAG

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

vivianecordeiro-sky commented on source/db/wifi_db_apis.c:
```diff
@@ -6681,18 +6698,18 @@ int wifidb_init_vap_config_default(int vap_index, wifi_vap_info_t *config,
         cfg.vap_mode = wifi_vap_mode_ap;
         if (isVapPrivate(vap_index)) {
             cfg.u.bss_info.showSsid = true;
+#ifdef FEATURE_SUPPORT_WPS
```
Can you check HAL code if the configuration can be reject if these parameter are not initialized and the vlues is invalid?

vivianecordeiro-sky commented on source/dml/tr_181/ml/cosa_wifi_dml.c:
```diff
@@ -11801,8 +11789,7 @@ WPS_SetParamBoolValue
     }
-
-    /* CcspTraceWarning(("Unsupported parameter '%s'\n", ParamName)); */
+#endif
```
add message for command not supported

vivianecordeiro-sky commented on source/dml/tr_181/ml/cosa_wifi_dml.c:
```diff
@@ -11846,348 +11833,328 @@ WPS_SetParamIntValue
     }
-
-    /* CcspTraceWarning(("Unsupported parameter '%s'\n", ParamName)); */
+#endif
```
add message for command not supported

vivianecordeiro-sky commented on source/dml/tr_181/ml/cosa_wifi_dml.c:
```diff
@@ -11846,348 +11833,328 @@ WPS_SetParamIntValue
+            }
+            return TRUE;
+        }
+#endif
```
add message for command not supported

vivianecordeiro-sky commented on source/webconfig/wifi_decoder.c:
```diff
@@ -1609,6 +1609,7 @@ webconfig_error_t decode_vap_common_object(const cJSON *vap, wifi_vap_info_t *va
     decode_param_bool(vap, "BssHotspot", param);
     vap_info->u.bss_info.bssHotspot = (param->type & cJSON_True) ? true : false;
 
+#ifdef FEATURE_SUPPORT_WPS
```
keep parameters in the webconfig subdoc

vivianecordeiro-sky commented on source/webconfig/wifi_encoder.c:
```diff
@@ -429,7 +429,7 @@ webconfig_error_t encode_vap_common_object(const wifi_vap_info_t *vap_info,
     // BssHotspot
     cJSON_AddBoolToObject(vap_object, "BssHotspot", vap_info->u.bss_info.bssHotspot);
-
+#ifdef FEATURE_SUPPORT_WPS
```
don't change this file

vivianecordeiro-sky reviewed (changes requested):
please address the comments

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

Commit:
Merge branch 'develop' into WPS_DISTRO_FLAG

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

SanjayVenkatesan190 committed:
Merge branch 'rdkcentral:develop' into WPS_DISTRO_FLAG

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com

Commit:
RDKB-59244: WPS Enhancement task

Impacted Platforms: All RDKB platforms.

Reason for change: Moved WPS code under separate CFLAGS to
enable for appropriate platforms.

Test Procedure:
1. Load the above mentioned build
2. If the loaded device supports WPS , then trigger the WPS push button
led turns blue and client would connect via WPS.
3. If the loaded device doesn't support WPS, then trigger the WPS push button
and verify the WPS didn't get triggered, if triggered then issue reproduced.

Risks: Medium

Priority: P1

Signed-off-by: sanjayvenkatesan1902@gmail.com
